src/univalue/univalue.cpp

```diff
@@ -175,21 +182,21 @@ bool UniValue::checkObject(const std::map<std::string,UniValue::VType>& t)
 const UniValue& UniValue::operator[](const std::string& key) const
 {
     if (typ != VOBJ)
-        return nullValue;
+        return NullUniValue;
 
     int index = findKey(key);
     if (index < 0)
-        return nullValue;
+        return NullUniValue;
 
     return values[index];
 }
 
 const UniValue& UniValue::operator[](unsigned int index) const
 {
     if (typ != VOBJ && typ != VARR)
-        return nullValue;
+        return NullUniValue;
     if (index >= values.size())
-        return nullValue;
+        return NullUniValue;
```

Comment:
I guess it will throw an exception if you call any of the ``get_*`` methods on a null result. Failing faster would be preferable though. It's unfortunate that the current code may depend on this behaviour, so we probably can't change it :-(